src/coreclr/nativeaot/Runtime/threadstore.cpp

```diff
@@ -110,9 +110,16 @@ void ThreadStore::AttachCurrentThread(bool fAcquireThreadStoreLock)
     // we want to avoid at construction time because the loader lock is held then.
     Thread * pAttachingThread = RawGetCurrentThread();
 
-    // The thread was already initialized, so it is already attached
+    if (pAttachingThread->IsDetached())
+    {
+        ASSERT_UNCONDITIONALLY("The thread has been detached already");
```

Comment:
Can we make the error message to be identical to the CoreCLR one? https://github.com/dotnet/runtime/pull/112809/files#diff-63935b2e5a324d25feff725f83200eaa716c69fdef7529d2bd3cfa35620105aeR1776

Reply:
> Can we make the error message to be identical to the CoreCLR one?

Yes, we can do that. 

Technically, with this change, a NativeAOT thread will not have a "destroyed" state. There is "detached" when we can't run managed code any more and then there is "thread is gone, because native TLS is gone".  It seems we do not need a state in between and things gets simpler without it.  

For the purpose of error message the difference is too subtle though.